src/dynamic.f90

```diff
@@ -342,10 +344,23 @@ subroutine md(env,mol,chk,calc, &
       endif
       if (metasetlocal%nstruc.eq.0) then
          do i = 1, mol%n
-            do j = 1, 3
-               call random_number(dum2)
-               metasetlocal%xyz(j,i,1) = mol%xyz(j,i) + 1.0e-6_wp*dum2
+            ! Generate randomly displaced geometry
+            do while(.true.)
+               rnorm = 0.0_wp
+               do j = 1, 3
+                  call random_number(rcoord(j))
```

Comment:
I think this short loop is unnecessary if you pass the whole `rcoord` array to `random_number`

Reply:
True, there's also an intrinsinc function for computing the norm. My FORTRAN is a bit rusty